src/sentry/static/sentry/app/utils/integrationUtil.tsx

```diff
@@ -40,8 +40,16 @@ export const isIntegrationDirectoryActive = (organization?: Organization) => {
   }
 };
 
-export const getSortIntegrationsByWeightActive = () =>
-  localStorage.getItem(SORT_INTEGRATIONS_BY_WEIGHT) === '1';
+export const getSortIntegrationsByWeightActive = (organization?: Organization) => {
+  const variant = organization?.experiments?.IntegrationDirectorySortWeightExperiment;
+  if (localStorage.getItem(SORT_INTEGRATIONS_BY_WEIGHT) === '1') {
```

Comment:
@NisanthanNanthakumar Wondering if a case statement might not be better:
```
switch(localStorage.getItem(SORT_INTEGRATIONS_BY_WEIGHT)) {
    case '1':
        return true;
    case '0':
        return false;
    default:
        return variant && variant === '1';
}
```
Take it as an optional suggestion :)